Clarify error for non-UTF-8 encoded data in runpath setup (#7141)
equinor/ert: docs/reference/configuration/keywords.rst

.. _data_file:
.. topic:: DATA_FILE

    Specify the filepath to the ``.DATA`` file of Eclipse/flow.
    This does two things:

    1. Template the ``DATA_FILE`` using :ref:`RUN_TEMPLATE <run_template>`.

        - The templated file will be named according to :ref:`ECLBASE <ECLBASE>` and copied to the runpath folder.
        - Note that support for parsing the Eclipse/flow data file is limited, and using explicit templating with :ref:`RUN_TEMPLATE <run_template>` is recommended where possible.
    2. Set :ref:`NUM_CPU <num_cpu>`

        - ert will search for ``PARALLEL`` in the data file, and it not found, it will search for ``SLAVE`` and update :ref:`NUM_CPU <num_cpu>` according to how many nodes are found.

>         - Note that it does *not* parse the data files of the nodes, and will assume one cpu per node where entry number 5 is not set, and the number of entry number 5 otherwise plus one cpu for the master node.

Author: I don't understand this. What is entry number 5?

.. _data_file:
.. topic:: DATA_FILE

>     Specify the filepath to the ``.DATA`` file of Eclipse/flow.

Reviewer: nit-picking: I believe `ECLIPSE` is usually capitalised and I would also use `OPM Flow` instead of flow

Author: Would you mind using the `Suggest` feature in GitHub so that I can just commit your proposed changes directly from GitHub?

>     Specify the filepath to the ``.DATA`` file of Eclipse/flow.

Reviewer: ```suggestion
    Specify the filepath to the ``.DATA`` file of ECLIPSE/OPM Flow.
```

    This does two things:

    1. Template the ``DATA_FILE`` using :ref:`RUN_TEMPLATE <run_template>`.

        The templated file will be named according to :ref:`ECLBASE <ECLBASE>` and

>         copied to the runpath folder. Note that support for parsing the Eclipse/flow

Reviewer: and here too

>         copied to the runpath folder. Note that support for parsing the Eclipse/flow

Reviewer: ```suggestion
        copied to the runpath folder. Note that support for parsing the ECLIPSE/OPM Flow
```

        data file is limited, and using explicit templating with :ref:`RUN_TEMPLATE
        <run_template>` is recommended where possible.

    2. Set :ref:`NUM_CPU <num_cpu>`

        ert will search for ``PARALLEL`` in the data file, and if not found, will
        search for ``SLAVES`` and update :ref:`NUM_CPU <num_cpu>` according to how
        many nodes are found.

>         The fifth entry of the keyword ``SLAVES`` determines the number of processors

Reviewer: This is a bad wording to me. The keyword ``SLAVES`` is defined through a set of up to 5 numbers:
```
SLAVES
--    slave   datafile    machine        directory                    number of
--    name    root        hostname      of data file                 processors
--
      'upper'  'base'       '*'        '/usr/models/upper'            1    /
      'lower'  'base'       '*'        '/usr/models/lower'            5   /

/

```

Author: What do you suggest?

Reviewer: and I read `fifth entry of the keyword ``SLAVES`` ` as if this keyword will be entered 5 times. But I am also not familiar with the vocabulary, so I might be wrong

>         The fifth entry of the keyword ``SLAVES`` determines the number of processors

Author: I think of mambo number five when I read that.

>         to run the slave on.

Reviewer: ```suggestion
        to run the slave reservoir on.
```

        Ert does not parse the data files of the nodes, and assumes one CPU per node
        where the fifth entry is not set.
        The number of CPUs available for the master node is the number specified in the

>         fifth entry of ``SLAVES`` plus one.

Reviewer: I do not think it is true - I read the documentation before as there will be one CPU node for master and whatever specified for slaves

Reviewer: I will double-check it now

Author: Not sure I understand. Do you mean that you read the previous version of the documentation differently than what I suggest here?

Reviewer: yes

Author: This is the previous version:

"... and the number of entry number 5 otherwise plus one cpu for the master node."
 
 I find this really hard to parse.
 What does the `plus` here refer to you think?

Reviewer: So after a brief discussion with @berland we agreed on the fact the the total number of CPUs needed is the sum of whatever is specified in the 5th item of each slave reservoir (with default 1) plus one CPU for the master process. I can come up with suggestion that reflects that

>         ert will search for ``PARALLEL`` in the data file, and if not found, will
>         search for ``SLAVES`` and update :ref:`NUM_CPU <num_cpu>` according to how
>         many nodes are found.
>         The entry number five of the keyword ``SLAVES`` determines the number of processors
>         to run the slave reservoir on.
>         Ert does not parse the data files of the nodes, and assumes one CPU per node
>         where the fifth entry is not set.
>         The number of CPUs available for the master node is the number specified in the
>         fifth entry of ``SLAVES`` plus one.

Reviewer: ```suggestion
                ert will search for ``PARALLEL`` in the data file, and if not found, will search for ``SLAVES`` and update :ref:`NUM_CPU <num_cpu>` according to how
        many nodes are found.
        The fifth item of the keyword ``SLAVES`` determines the number of processors
        to run the slave reservoir on with the default value of `1` when the fifth item is not set.
        Note that ert does not parse the data files directly and relies on what is specified in the ``SLAVES`` keyword.
        Therefore, the total number of CPUs is the sum of the number of processes for each slave reservoir plus one for the master process.
```

Author: Real nice 👍

    Example:

    .. code-block::

        -- Load the data file called ECLIPSE.DATA
        DATA_FILE ECLIPSE.DATA

    See the :ref:`DATA_KW <data_kw>` keyword which can be used to utilize more template

>     functionality in the Eclipse/flow datafile.

Reviewer: ```suggestion
    functionality in the ECLIPSE/OPM Flow datafile.
```
